src/main/java/teammates/common/datatransfer/FeedbackQuestionAttributes.java

```diff
@@ -510,14 +509,10 @@ public void sanitizeForSaving() {
     
     public boolean isJsonValid(String jsonString) {
         try {
-            // Checks whether JSONArray is valid
-            new JSONObject(jsonString);
-        } catch (JSONException e) {
-            try {
-                new JSONArray(jsonString);
-            } catch (JSONException ex) {
-                return false;
-            }
+            JsonParser parser = new JsonParser();
+            parser.parse(jsonString);
```

Comment:
On a related note, we need unit tests to ensure the code works for both old format and new format. i.e. There should be a test failing if this method is broken.